subprojects/snapshots/src/test/groovy/org/gradle/internal/snapshot/impl/DirectorySnapshotterTest.groovy

```diff
@@ -178,6 +179,25 @@ class DirectorySnapshotterTest extends Specification {
         rootDir.listFiles()*.makeReadable()
     }
 
+
+    @Requires(TestPrecondition.UNIX_DERIVATIVE)
+    @Issue("https://github.com/gradle/gradle/issues/2552")
+    def "named pipe snapshots to MissingFileSnapshot"() {
+        def rootDir = tmpDir.createDir("root")
+        def pipe = rootDir.file("testPipe").createNamedPipe()
+
+        when:
+        def snapshot = directorySnapshotter.snapshot(rootDir.absolutePath, directoryWalkerPredicate(new PatternSet()), new AtomicBoolean(false))
+        then:
+        assert snapshot instanceof DirectorySnapshot
+        snapshot.children.collectEntries { [it.name, it.class] } == [
+            testPipe: MissingFileSnapshot
+        ]
+
+        cleanup:
+        pipe.delete()
```

Comment:
The `TestNameTestDirectoryProvider` already deletes all the files in the temporary directory. Why do you need to add this extra cleanup?

Reply:
I talked to @paplorinc, and we decided to leave the extra cleanup since the pipe is a special file. So we avoid any problems with the automatic cleanup and other tools.